src-utils/HydraAuctionUtils/Hydra/Runner.hs

```diff
@@ -130,12 +130,14 @@ instance MonadTrace HydraRunner where
 
 instance MonadBlockchainParams HydraRunner where
   queryBlockchainParams = do
-    MkHydraExecutionContext {fakeBlockchainParams} <- ask
-    return fakeBlockchainParams
+    MkHydraExecutionContext {l1Context} <- ask
+    params <- liftIO $ executeL1Runner l1Context queryBlockchainParams
+    protocolParameters <- liftIO readHydraNodeProtocolParams
+    return $ params {protocolParameters}
 
-  -- FIXME: Hydra started to support time but we still do not
-  convertValidityBound _ =
-    return (TxValidityNoLowerBound, TxValidityNoUpperBound)
+  convertValidityBound x = do
+    MkHydraExecutionContext {l1Context} <- ask
+    liftIO $ executeL1Runner l1Context $ convertValidityBound x
```

Comment:
Does hydra not provide a way to query this directly?

Reply:
No, they do not provide a way, there is no way to get SlotNo from Hydra API now.

See: https://github.com/input-output-hk/hydra/pull/858#issuecomment-1563028483